test/clj/chronograph/handlers/organization_test.clj

```diff
@@ -109,3 +113,93 @@
                (organization/find-one {:params {:slug slug}
                                        :user other-user})))
           "Fetching org details by a user that does not belong to the org fails with HTTP error."))))
+
+(deftest show-members-test
+  (testing "Should fetch joined and invited members and return a 200"
+    (let [user         (factories/create-user)
+          {:organizations/keys [slug]
+           org-id              :organizations/id} (factories/create-organization (:users/id user))
+          joined-user  (factories/create-user)
+          invited-user (invite/find-or-create! db/datasource org-id "invite@user.com")]
+      (acl/create! db/datasource {:user-id         (:users/id joined-user)
+                                  :organization-id org-id
+                                  :role            acl/member})
+      (is (= {:status 200
+              :body   {:invited [invited-user]
+                       :joined  [user joined-user]}}
+             (-> (organization/show-members {:params {:slug slug}
+                                             :user   user})
+                 (select-keys [:status :body]))))))
+
+  (testing "Should return a 404 if no org is found with the given slug"
+    (tu/with-fixtures [fixtures/clear-db]
```

Comment:
If `clear-db` is the only fixture we really need, perhaps it's better to break out `testing` blocks into their own deftests?

Alternatively, why not call `truncate-all-tables!` directly, if we want to retain the current `testing` blocks? The `with-fixtures` macro feels like a roundabout way of doing the same thing.

Also, minor nit: semantically, clojure.test defines fixtures as things attached to a namespace, which the runner is aware of.

Reply:
The problem with splitting tests into different `deftest`s purely for the sake of fixtures is that we lose out on the grouping that `deftest` and `testing` blocks provide naturally. It's quite burdensome when there are many tests in a namespace. Also we'll have to come up with a meaningful name for each `deftest`, which is difficult and has little value.

The benefit of using `with-fixtures` is that we can appy any fixture we choose, not just `clear-db`.

Reply:
I agree about the trouble of splitting deftests. I won't contest the macro. However, I'll offer the following commentary:

I'm not convinced the macro adds enough value for this use case, because simpler alternatives are easily doable, IMO. The macro here also needs work to extend to teardowns between `testing` blocks.

I'd be fine doing either of the following, with the understanding that the testing block already clarifies the intended scope of the effects. 

Alternative 1: Here I'd directly use the fns that our fixtures wrap over:

```
(testing "foo bar"
  (clear-db!)
  (setup-x!)
  (setup-y!)
 
  (is (... ))
  (is (... ))
  (is (... ))

  (teardown-x!)
  (teardown-y!)
  (teardown-z!))
```

Alternative 2: If we want to use the fixtures themselves, a HOF could work as well:
```
(defn exec-fixtures!
  [& fixtures]
  (reduce #(%2 %1)
          (fn [])
          fixtures))

;;;; and then 

(testing "something"
   (exec-fixtures! setup-fixture1
                   setup-fixture2
                   ... ...
                   setup-fixtureN)

    (is ( .... ))
    (is ( .... ))
    (is ( .... ))

   (exec-fixtures! teardown-fixture1
                   teardown-fixture2
                   ... ...
                   teardown-fixtureN))
```

Reply:
The teardown can be written in the fixtures themselves. For example:
```
 (defn fixture1
   [f]
   (setup1!)
   (f)
   (teardown1!))

 (defn fixture2
   [f]
   (setup2!)
   (f)
   (teardown2!))

 ;; ......

 (testing "something"
   (tu/with-fixtures [fixture1 fixture2]
     (is (...))
     (is (...))))
```
Do you still feel that the macro is not a big value-add?

Reply:
Ah, I did not consider the third alternative. Cool :+1: This actually makes the macro a better idea because it removes ordering problems between setup/teardown.